engine/clients/pgvector/upload.py

```diff
@@ -27,10 +32,28 @@ def upload_batch(
         vectors = np.array(vectors)
 
         # Copy is faster than insert
-        with cls.cur.copy("COPY items (id, embedding) FROM STDIN") as copy:
+        with cls.cur.copy(
+            "COPY items (id, embedding) FROM STDIN WITH (FORMAT BINARY)"
+        ) as copy:
+            copy.set_types(["integer", "vector"])
             for i, embedding in zip(ids, vectors):
                 copy.write_row((i, embedding))
 
+    @classmethod
+    def post_upload(cls, distance):
+        try:
+            hnsw_distance_type = cls.DISTANCE_MAPPING[distance]
+        except KeyError:
+            raise IncompatibilityError(f"Unsupported distance metric: {distance}")
+
+        cls.conn.execute("SET max_parallel_workers = 128")
+        cls.conn.execute("SET max_parallel_maintenance_workers = 128")
+        cls.conn.execute(
+            f"CREATE INDEX ON items USING hnsw (embedding {hnsw_distance_type}) WITH (m = {cls.upload_params['hnsw_config']['m']}, ef_construction = {cls.upload_params['hnsw_config']['ef_construct']})"
+        )
```

Comment:
Hm, might be a bit unfair to the other competitors, which also support loading in a "split-mode", however everyone is welcome to come and improve the setup if they think it is not optimised 🤷‍♂️